src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx

```diff
@@ -559,6 +570,22 @@ export class DashboardAppController {
       })
     );
 
+    // TODO: find nicer solution for this
```

Comment:
Suggestion: Make changing the URL a concern of the state manager - then at least it's hidden away in one place and this detail doesn't leak into the controller

Reply:
Done